packages/dds/sequence/src/test/reentrancy.spec.ts

```diff
@@ -114,6 +118,70 @@ describe("SharedString op-reentrancy", () => {
 			}
 		});
 
+		it("is empty after deleting reference pos in reentrant callback", () => {
```

Comment:
FWIW I think the work item here wasn't talking about op reentrancy. I interpreted it as issues with `LocalReferencePosition.link` and `LocalReferenceCollection.removeLocalRef` mutually calling each other, leading to issues.

Not really against these tests though, I guess they do add some coverage for actions taken as part of a sequence delta callback, which is a reasonable thing applications might do. I think we have some coverage of this type of thing at the merge-tree level but not really at the sequence level.